crates/ironrdp-pdu/src/cursor.rs

```diff
@@ -45,6 +45,12 @@ impl<'a> ReadCursor<'a> {
         self.pos
     }
 
+    #[inline]
+    pub fn discard(&mut self) {
+        let remaining_len = self.len();
+        self.advance(remaining_len);
+    }
```

Comment:
Issue: I feel mixed about this. My understanding is that this method is making two assumptions:

1) the cursor inner slice only ever contains a single PDU, and
2) it’s fine to skip the rest of the PDU no matter from where this method is called.

For 1), we typically never include more than one PDU per slice, so it should be generally fine, but following code will stop working as soon as `Pdu1` is relying on this method.
```
let some_slice = &[/* pdu1 payload */, /* pdu2 payload */];
let mut cursor = ReadCursor(some_slice);
let pdu1: Pdu1 = decode(&mut cursor)?;
let pdu2: Pdu2 = decode(&mut cursor)?;
```
It’s not something I would expect to see outside of tests, so it’s probably not too bad.

For similar reasons 2) is fine as long as there is no other item coming after. For instance decoding a list of item where one of the item could discard the rest of the payload is a problem.

I.e: the more layers, the more something could go wrong at unexpected places.

Overall, we should be very careful when using this method and I suggest we recommend against liberally using it. At the very least, a doc comment should explain all that.